zrml/prediction-markets/src/lib.rs

```diff
@@ -208,10 +256,28 @@ mod pallet {
                 market.status == MarketStatus::Reported || market.status == MarketStatus::Disputed,
                 Error::<T>::InvalidMarketStatus,
             );
-            Self::clear_auto_resolve(&market_id)?;
+            let (ids_len, disputes_len) = Self::clear_auto_resolve(&market_id)?;
             let market = T::MarketCommons::market(&market_id)?;
-            let weight = Self::on_resolution(&market_id, &market)?;
-            Ok((Some(weight), Pays::No).into())
+            let weight_on_resolution = Self::on_resolution(&market_id, &market)?;
+            // Substracted by internal_resolve_scalar_...,
+            // because it's returned in `on_resolution` and the benchmark of
+            // `admin_move_market_to_resolved_overhead_...` was initialised with a scalar market
```

Comment:
I'm very concerned about this approach. This seems hard to maintain, hard to understand and makes the code difficult to read. There are two alternatives:

- Benchmark each function like `admin_move_market_to_resolved` by identifying each of its execution paths and benchmarking each path separately
- Use dependency injection to inject a noop `on_resolution`

Both are fine in my mind, although the latter seems to subordinate the structure of the code to the benchmarks, which is why I would actually prefer the first approach.

Reply:
Ok, but then I would like to wait for https://github.com/zeitgeistpm/zeitgeist/pull/813 to get merged.

Reply:
https://github.com/zeitgeistpm/zeitgeist/pull/812/commits/32f9ca233831fca3654894158335e9dce31c19c1